### ISSUE TITLE
Installer hangs indefinitely when installing on 1-core VM

### ISSUE THREAD

**Issue by mtornwall** [Starcounter/Starcounter#2759]
When attempting to install Starcounter on an Azure VM with 1 core and 1.75 GiB memory running Windows Server 2012 R2 the installer hangs indefinitely on "Checking system requirements...".

I'm aware that Starcounter requires two cores, but I think this should be reported as an error instead.

I can provide a VM for testing if required.

Update: Switching to a VM configuration with two cores resolved the problem.

**Comment by alemoi** [comment]
Thanks for reporting! Definitely should not hang like that! Let us test this case and see what's going on!

**Comment by mtornwall** [comment]
Let me know if you want me to set up a VM for you.

**Comment by warpech** [comment]
The above closed issue has information how to reproduce it in Parallels (incl screenshot).

Note that insufficient RAM is detected correctly, with an error message:

> For being productive Starcounter recommends that your machine has at least 4Gb of RAM.
You can now proceed with installation...

**Comment by alemoi** [comment]
Thanks for additional information!
I have found the reason for this, its inside installer GUI. The following code as it should throws an exception about 1 core limitation:
```cs
// Checking number of cores.
            if (Environment.ProcessorCount <= 1)
            {
                throw ErrorCode.ToException(Error.SCERRINSTALLERABORTED,
                    "To run Starcounter you must have more than one logical core on the machine.");
            }
```
however the problem occurs when the following code catches and processes the exception:
```c#
catch (Exception e) {
	
	// Error / Message
	this._dispatcher.BeginInvoke(DispatcherPriority.Normal,
		new Action(delegate {
		this.OnError(e);
	}
	));
}
```
If I replace the exception processing to the following, then everything is fine:
```c#
catch (Exception e) {
	MessageBox.Show(e.ToString());
}
```

I guess something wrong happens in `this.OnError` in GUI installer, it never shows error dialog about 1 core requirement. 

@andwah can you please take a look, assigning to you so far.

**Comment by alemoi** [comment]
Another problem that I found now is that even after installation on this Windows 10 machine, all Starcounter setup processes remain running: GUI installer process does not terminate and other processes are waiting for it.

@andwah sorry but its another thing for you to check, why GUI installer process does not terminate after installation is finished, maybe smth with WPF?

By the way, I used latest VirtualBox and https://dev.modern.ie/tools/vms/windows/ (MSEdge on Win10 with VirtualBox)

**Comment by warpech** [comment]
@andwah any progress with checking this?

**Comment by alemoi** [comment]
@andwah found two reasons for this behavior. One was missing CRT libraries during temp extraction (fixed via 65bb5a730492aad1a99cb8c3172c51a0fea660c7) and another is that exceptions are not expected in statistics send code, which @andwah will fix.

**Comment by andwah** [comment]
The installer was waiting for a callback, It was never called when there was an exception.
It as been fixed here f5ef02f15091081b2913ec6f89985c9500e5d20a
